static/src/javascripts-legacy/projects/common/utils/check-mediator.js

```diff
@@ -0,0 +1,146 @@
+     * Each object contains 3 fields: id (string), canRun (boolean), dependentChecks (object)
+     * dependentChecks should contain 2 fields: passCondition (SOMECHECKSPASSED/EVERYCHECKPASSED), list (nested array of checks)
+     * If object has dependentChecks then the DefferedCheck will resolve when these dependentChecks have all resolved
+     *
+    **/
+
+    var checkList = [{
+            id: 'isOutbrainNonCompliant',
+            dependentChecks: {
+                passCondition: SOMECHECKSPASSED,
+                list:[{
+                    id: 'isUserInAClashingAbTest'
+                }, {
+                    id: 'canEmailBeInserted',
+                    dependentChecks: {
+                        passCondition: EVERYCHECKPASSED,
+                        list: [{
+                            id: 'emailCanRun'
+                        }, {
+                            id: 'listCanRun'
+                        }, {
+                            id: 'emailInArticleOutbrainEnabled'
+                        }]
+                    }
+                }]
+            }
+        }];
+
+    function DeferredCheck(dependentCheckPromises, dependentChecksPassCondition) {
+        this.complete = new Promise(function(resolve, reject) {
+            this.resolve = resolve;
+            this.reject = reject;
+        }.bind(this));
+
+        if (dependentCheckPromises) {
+            Promise.all(dependentCheckPromises).then(function(results) {
+
+                var hasPassed = function(result) {
+                    return result;
+                };
+
+                if (dependentChecksPassCondition.call(results, hasPassed)) {
+                    this.resolve(true);
+                } else {
+                    this.resolve(false);
+                }
+            }.bind(this));
+        }
+    }
+
+    function registerDefferedCheck(check) {
+        var dependentCheckPromises = [];
+
+        if (check.dependentChecks) {
+            check.dependentChecks.list.forEach(registerDependentCheck.bind(null, dependentCheckPromises));
+        
+            return new DeferredCheck(dependentCheckPromises, check.dependentChecks.passCondition);
+        }
+
+        return new DeferredCheck();
+    }
+
+    
+    function registerDependentCheck(dependentCheckPromises, dependentCheck) {
+        dependentCheckPromises.push(registerCheck(dependentCheck).complete);
+    }    
+
+    function registerCheck(check) {
+        var registeredCheck = registerDefferedCheck(check);
+
+        registeredChecks[check.id] = registeredCheck;
+
+        return registeredCheck;
+    }
+
+    /**
+     * public
+    **/
+    function init() {
+        checkList.forEach(registerCheck);
+    }
+    
+    function resolveCheck(id) {
+        var argsArray = Array.prototype.slice.call(arguments, 1);
+        
+        if (registeredChecks[id]) {
+            return registeredChecks[id].resolve.apply(null, argsArray);
+        }
+    }
+
+    function rejectCheck(id) {
```

Comment:
Yep I created and exposed this when setting the module up but you're correct it's not used as promises should be resolved with true/false, so yes this can be removed.